### PR TITLE
Doc updates for aztables

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Detail behavior when deleting a table.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/10813

File: sdk/data/aztables/client.go [hunk 1]
```diff
@@ -653,7 +658,7 @@ func (s *SetAccessPolicyOptions) toGenerated() *generated.TableClientSetAccessPo
 // Specify nil for options if you want to use the default options.
 func (t *Client) SetAccessPolicy(ctx context.Context, options *SetAccessPolicyOptions) (SetAccessPolicyResponse, error) {
```
Comment (jhendrixMSFT) [inline sdk/data/aztables/client.go:659]: While fixing the panic, I wonder if `SetAccessPolicyOptions` should actually be optional?  Would one ever call this API with a `nil` options?

Comment (jhendrixMSFT) [reply sdk/data/aztables/client.go:659]: To clarify, should `TableACL` be optional?

File: sdk/data/aztables/client.go [hunk 2]
```diff
@@ -128,6 +128,8 @@ func (t *Client) CreateTable(ctx context.Context, options *CreateTableOptions) (
 
 // Delete deletes the table with the tableName specified when NewClient was called. If the service returns a non-successful HTTP status
 // code, the function returns an *azcore.ResponseError type. Specify nil for options if you want to use the default options.
+// NOTE: deleting a table can take up to 40 seconds or more to complete.  If a table with the same name is created while the delete is still
+// in progress, an *azcore.ResponseError is returned with error code TableBeingDeleted and status code http.StatusConflict.
```
Comment (jhendrixMSFT) [inline sdk/data/aztables/client.go:132]: Unfortunately, the only operation that returns a 409 is creating the table.  Other operations return a 404 so you can't spin on them to deterministically tell when the table has been deleted.  It means that callers of `CreateTable` would need to be resilient to this 409 case.

Comment (jhendrixMSFT) [reply sdk/data/aztables/client.go:134]: Added doc comment to `CreateTable` explaining this.

File: sdk/data/aztables/client.go (hunk 1 above)
Comment (seankane-msft) [reply sdk/data/aztables/client.go:661]: That's a good point. TableACL probably should not be optional. I'm surprised this didn't come up in archboard. I wonder if by specifying nil you can remove access policies. Let me investigate.

Comment (jhendrixMSFT) [reply sdk/data/aztables/client.go]: I've removed the other changes so this PR is just doc updates now.

**Review by seankane-msft — APPROVED** [review]

**Pull request merged**